presto-hive/src/main/java/com/facebook/presto/hive/HiveCompressionCodec.java

```diff
@@ -20,24 +20,44 @@
 import parquet.hadoop.metadata.CompressionCodecName;
 
 import java.util.Optional;
+import java.util.function.Predicate;
 
+import static com.facebook.presto.hive.HiveStorageFormat.DWRF;
+import static com.facebook.presto.hive.HiveStorageFormat.ORC;
 import static java.util.Objects.requireNonNull;
 
 public enum HiveCompressionCodec
 {
-    NONE(null, CompressionKind.NONE, CompressionCodecName.UNCOMPRESSED),
-    SNAPPY(SnappyCodec.class, CompressionKind.SNAPPY, CompressionCodecName.SNAPPY),
-    GZIP(GzipCodec.class, CompressionKind.ZLIB, CompressionCodecName.GZIP);
+    NONE(null, CompressionKind.NONE, CompressionCodecName.UNCOMPRESSED, allStorageFormatsSupported().negate()),
```

Comment:
nit: f -> false might be shorter and easier to read

Reply:
I agree `allStorageFormatsSupported().negate()` might be a bit difficult to follow, but I guess @jainxrohit 's initial motivation not to  inline the lambda  is it might help a little bit with the intention (basically, the method name is used as comment). 

An alternative way would be have a separate method `noneStorageFormatsSupported`. No strong opinion though :)

Reply:
It seems like the general opinion seems to be on the side of using lamdas directly. I would change it.